main.cpp

```diff
@@ -42,16 +42,61 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
+
+template<typename Type>
+struct Wrapper
+{
+    Wrapper(Type&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    Type val;
+
+    void print()
+    {
+        std::cout << val << std::endl;
+    }
+};
+
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    Point val;
+
+    void print()
+    {
+        std::cout << val.toString() << std::endl;
+    }
+};
+
+
+
+template<typename T, typename ...Args>
+void variadicHelper(T first, Args ... everythingElse)
+{
+    Wrapper<T> tWrapper(std::forward<T>(first));
+    tWrapper.print();
+
+    //while(sizeof...everthingElse != 0)
+    variadicHelper( everythingElse... ); //recursive call
+}
+
+template<typename T>
+void variadicHelper(T last)
+{
+    Wrapper<T> tWrapper(std::forward<T>(last));
+    tWrapper.print();
+}
+
```

Comment:
Project9.1 challenge 2:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?

Reply:
How do I denote an empty variadic parameter list? And do I make a variadicHelper func that takes only a parameter pack?